### PR TITLE
CORTX-31409: Jenkins Job for whitesource container scan 

### PR DESCRIPTION
**Pull request by Abhirup-A-G**
Signed-off-by: Abhirup Ghosh <abhirup.ghosh@seagate.com>

# Problem Statement
- CORTX-31409: Jenkins Job for whitesource container scan 

# Design
-  https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/1018429475/DEV-10+Pilot+implementation+for+container+scan+approach+Whitesource
- New Jenkins Job after code change as per review : https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/AbhirupSpace/job/whitesource_scan_job_test_6/5/
# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [x] Testing was performed with RPM
- [x] Jenkins pipelines used for testing

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/security/whitesource_container_scan.groovy [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        string(name: 'SNS_CONFIG', defaultValue: '1+0+0', description: 'sns configuration for deployment. Please select value based on disks available on nodes.', trim: true)
+        string(name: 'DIX_CONFIG', defaultValue: '1+0+0', description: 'dix configuration for deployment. Please select value based on disks available on nodes.', trim: true)
+        string(name: 'CONTROL_EXTERNAL_NODEPORT', defaultValue: '31169', description: 'Port to be used for control service.', trim: true)
+        string(name: 'S3_EXTERNAL_HTTP_NODEPORT', defaultValue: '30080', description: 'HTTP Port to be used for IO service.', trim: true)
+        string(name: 'S3_EXTERNAL_HTTPS_NODEPORT', defaultValue: '30443', description: 'HTTPS to be used for IO service.', trim: true)
+        string(name: 'NAMESPACE', defaultValue: 'cortx', description: 'kubernetes namespace to be used for CORTX deployment.', trim: true)
+        text(defaultValue: '''hostname=<hostname>,user=<user>,pass=<password>''', description: 'VM details to be used. First node will be used as Primary node', name: 'hosts')
+        choice(
+            name: 'EXTERNAL_EXPOSURE_SERVICE',
+            choices: ['LoadBalancer', 'NodePort'],
```
[the remark this answers is not in the document]
Comment (Abhirup-A-G) [reply jenkins/security/whitesource_container_scan.groovy:35]: ok.done

File: jenkins/security/whitesource_container_scan.groovy [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+                        cat hosts
+                        export SOLUTION_CONFIG_TYPE=automated
+                        export WHITESOURCE_SERVER_URL=${WHITESOURCE_SERVER_URL}
+                        export API_KEY=${API_KEY}
+                        export USER_KEY=${USER_KEY}
+                        export PRODUCT_NAME=${PRODUCT_NAME}
+                        export DOCKER_REGISTRY=${DOCKER_REGISTRY}
+                        export WHITESOURCE_VERSION=${WHITESOURCE_VERSION}
+                        export PULL_SECRET=${PULL_SECRET}
+                        ./whitesource_container_scan_function.sh
```
Comment (shailesh-vaidya) [inline jenkins/security/whitesource_container_scan.groovy:116]: Rename `whitesource_container_scan_function.sh` as `whitesource_container_scan.sh` and vice versa. Execute `whitesource_container_scan.sh` through Jenkins job.

File: scripts/security/whitesource_container_scan_function.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    export API_KEY=$API_KEY && 
+    export USER_KEY=$USER_KEY && 
+    export PRODUCT_NAME=$PRODUCT_NAME && 
+    export DOCKER_REGISTRY=$DOCKER_REGISTRY && 
+    export WHITESOURCE_VERSION=$WHITESOURCE_VERSION && 
+    export PULL_SECRET=$PULL_SECRET && /var/tmp/whitesource_container_scan.sh"
+}
+
+#Execution
+script_execution
```
Comment (shailesh-vaidya) [inline scripts/security/whitesource_container_scan_function.sh:46]: This function is only calling the above command so we can call them directly i.e. without function.

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: jenkins/security/whitesource_container_scan.groovy (hunk 2 above)
Comment (Abhirup-A-G) [reply jenkins/security/whitesource_container_scan.groovy]: Done

File: scripts/security/whitesource_container_scan_function.sh (hunk 3 above)
Comment (Abhirup-A-G) [reply scripts/security/whitesource_container_scan_function.sh]: Done

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
Please fix Codacy issues. 
Please execute the Jenkins job with the latest changes and add URL.

**Review by shailesh-vaidya — APPROVED** [review]

**Comment by shailesh-vaidya** [comment]
@gauravchaudhari02  - Please review

**Review by gauravchaudhari02 — APPROVED** [review]

**Pull request merged**